modules/screenshot.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: UTF-8 -*-
+#
+#       screenshot.py
+#
+#       Copyright CherryTree 2009-2017 Giuseppe Penone <giuspen@gmail.com>
+#
+#       Copyright screenshot.py 2018 David Holland <davidholland5499@outlook.com>, as long as it is a standalone module and not integrated with Cherry Tree
+#
+#       This program is free software; you can redistribute it and/or modify
+#       it under the terms of the GNU General Public License as published by
+#       the Free Software Foundation; either version 3 of the License, or
+#       (at your option) any later version.
+#
+#       This program is distributed in the hope that it will be useful,
+#       but WITHOUT ANY WARRANTY; without even the implied warranty of
+#       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#       GNU General Public License for more details.
+#
+#       You should have received a copy of the GNU General Public License
+#       along with this program; if not, write to the Free Software
+#       Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+#       MA 02110-1301, USA.
+
+import gtk
+import gtk.gdk
+
+class ScreenshotWindow(gtk.Window):
+    window_modifier = "move"
+    window_mode = "Move-Mode"
+    window_instructions = '<span size="xx-large">To switch between Move- and Resize-Mode,\nsimply press <i>Space</i> or <i>Tab</i>.\nIf you want to move the upper, lower, left or right edge individually\npress either the <i>arrow</i> or <i>WASD</i> keys to switch modes.\nTo return to Move-Mode again press <i>space</i> or <i>tab</i>!\nIf you are in the desired mode,\nclick and drag with the left mouse button to modify the window.\nIf you are happy with the screen snippet,\npress the <i>Return</i> key to take the screenshot,\nor press the <i>Esc</i> key to cancel the screenshot.\n\n<b>Press Space to dismiss this dialog!</b></span>'
+
+    pixbuf_result = None
+    
+    label = gtk.Label()
+    # label.set_justify(gtk.JUSTIFY_CENTER)
+
+    def __init__(self, ready=None):
+        gtk.Window.__init__(self)
+        width = gtk.gdk.screen_width()
+        height = gtk.gdk.screen_height()
+
+        # self.connect("destroy",lambda wid:gtk.main_quit())
+        self.move(250,100)
+        
+        self.resize(width-500, height-200)
+        self.set_opacity(0.7)
+        self.set_keep_above(True)
+        self.set_decorated(False)
+        
+        self.connect("key_press_event", self.on_key_press)
+        self.connect("key_release_event", self.on_key_release)
+        self.connect("button-press-event", self.on_clicked)
+        self.set_events(gtk.gdk.KEY_PRESS_MASK | gtk.gdk.KEY_RELEASE_MASK | gtk.gdk.BUTTON_PRESS_MASK)
+        
+        col = gtk.gdk.Color('#666')
+        self.modify_bg(gtk.STATE_NORMAL, col)
+
+        self.label.show()
+        
+        self.add(self.label)
+        
+        self.update_mode_label(self.label)
+
+        self.fullscreen()
+
+    def run(self):
```

Comment:
Dialog-like function that blocks all activity (but does not freeze the GUI), until it is finished or aborted